rasa/nlu/featurizers/sparse_featurizer/regex_featurizer.py

```diff
@@ -248,7 +248,7 @@ def load(
                 finetuning.
             **kwargs: Any other arguments.
         """
-        file_name = meta.get("file")
+        file_name = meta["file"]
```

Comment:
Should we also raise an error if file is not in the metadata (similar to [here](https://github.com/RasaHQ/rasa/pull/9062/files#diff-b75ba088655bba2a10f267458c7ce01a1d2b9d69e4447f3f548bbf00db66d208R1172-R1177))?

Reply:
I suppose we could do that for every component. But then we also need tests for this otherwise we'll run into coverage problems. And I am thinking it's kind of a niche scenario... I mean to get there: the model zip is still intact, the overall json structure of the metadata is still intact, buuuuut the file key disappeared. I don't think that is really likely absent of willful user manipulation.

Reply:
Right now I am more in favor of dropping that new exception class again XD

Reply:
@wochinge what do you think?

Reply:
How come you added it for diet? Is it more likely there? (genuine question). I'm generally in favour of the symmetry, but understand how that blows up the scope, so I trust your judgment on this one 🤷‍♀️

Reply:
Yeah it's true it should be consistent. I added it there because Tobias saw it there and I/we didn't realize that it would be the same for all the other components as well. Now I am wondering whether it is a good idea or not

Reply:
I think I'll create a follow up issue for testing the consistency of metadata dicts - I think this also needs a bit of thought, e.g. whether we we would introduce a more specific type or not for the metadata. Wouldn't want to just produce a lot of code as a quick fix now.

Reply:
I created the issue and also changed the improper uses of get in the component load classes. Sometimes these usages would just push the error further back and show up as a `None` in a method that should have gotten the file name string. Now things crash directly where the problem is: that `file` is not in the metadata.

Reply:
Sounds good to me. Thanks for creating the issue!